CRM/Core/Payment.php

```diff
@@ -1277,9 +1277,12 @@ public function doPayment(&$params, $component = 'contribute') {
    *
    * @param array $params
    *
+   * @return array
    * @throws \Civi\Payment\Exception\PaymentProcessorException
    */
-  public function doRefund(&$params) {}
+  public function doRefund(&$params) {
+    return [];
```

Comment:
what do we want to return? I'm keen to go with an exception is a failure & otherwise it's success because adding payment_status_id to the  array always felt hacky & we don't need to  return an array AND pass by ref

Reply:
See follow-on in #15479.  I've kept them separate in terms of discussion that might be required :-)